core-bundle/src/Filesystem/HashGeneratorInterface.php

```diff
@@ -22,8 +22,12 @@ interface HashGeneratorInterface
      * If a value for $lastModified is provided, it is allowed to return null
      * instead of a hash. Doing this indicates that the hash has not changed in
      * the meantime and does not need to be recomputed.
+     *
+     * If you know the last modified timestamp (e.g. because you read it from
+     * the filesystem), assign the value to &$fileLastModified. This will prevent
+     * the calling code from reading it (again).
      */
-    public function hashFileContent(FilesystemAdapter $filesystem, string $path, int $lastModified = null): ?string;
+    public function hashFileContent(FilesystemAdapter $filesystem, string $path, int $lastModified = null, int &$fileLastModified = null): ?string;
```

Comment:
My gut feeling tells me, that this method might get another property or return value for some reason we do not know now. I'd thus propose to use a data object as argument and return value. It would allow us to easily extend the object with another property and we wouldn't have any reference handling here.

Reply:
Had this same feeling when writing the code and tried to avoid another object. :see_no_evil:  But I'm already inclined to adding one.

Reply:
I implemented a result object (context) in fe99dfa520c5a43463cc4089be5afdb15a1a4239.